src/session/parse_enum.rs

```diff
@@ -0,0 +1,63 @@
+use crate::session::UnknownEnumValueError;
+
+pub trait ProtoEnum<Proto> {
+    // Returns a default value for given Protobuf definition.
+    fn default() -> Proto;
+
+    // Returns value of the enum or None if the input i32 does not describe any know enum value.
+    // ::prost::Enumeration cannot be used instead, because it's not a trait.
+    fn from_i32(val: i32) -> Option<Proto>;
+}
+
+// Maps the raw integer value to enum value or returns ParseError when the value cannot be mapped.
+pub fn parse_enum<T: ProtoEnum<T>>(raw_enum_value: Option<i32>) -> Result<T, UnknownEnumValueError> {
+    match raw_enum_value {
+        Some(int_value) => match T::from_i32(int_value) {
+            Some(parsed_value) => Ok(parsed_value),
+            None => Err(UnknownEnumValueError::new(
+                std::any::type_name::<T>(),
+                int_value,
+            )),
+        },
+        None => Ok(T::default()),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[derive(Debug, PartialEq, ::prost::Enumeration)]
+    pub enum TestEnum {
+        One = 1,
+        Two = 2,
+    }
+
+    impl ProtoEnum<TestEnum> for TestEnum {
+        fn default() -> TestEnum { TestEnum::Two }
+        fn from_i32(val: i32) -> Option<TestEnum> {
+            TestEnum::from_i32(val)
+        }
+    }
+
+    #[test]
+    fn parse_empty_enum_test() {
+        let empty: Option<i32> = None;
+        let parsed : TestEnum = parse_enum(empty).unwrap();
```

Comment:
There should be no space before colon. Also, I don't think you need any type annotations here (and in these tests in general).